src/offsetPositionAnchor.js

```diff
@@ -18,18 +23,107 @@
+  }
+
+  function parseTwoValuePosition (first, second) {
+    if (first === 'top' || first === 'bottom' || second === 'left' || second === 'right') {
+      // Normalize so left|right appear first and/or top|bottom appear last.
+      // The position is only valid if there is one vertical value and one
+      // horizontal value, and both are keywords (not length-percentage).
+      if (!internalScope.isInArray(['top', 'center', 'bottom'], first) ||
+          !internalScope.isInArray(['left', 'center', 'right'], second)) {
+        return undefined;
+      }
+      // We received [ top | center | bottom ] [ left | center | right ]
+      return parseTwoValuePosition(second, first);
+    }
+
+    // We have
+    // [ left | center | right | <length-percentage> ]
+    // [ top | center | bottom | <length-percentage> ]
+
+    if (first === 'left') {
+      first = '0%';
+    } else if (first === 'center') {
+      first = '50%';
+    } else if (first === 'right') {
+      first = '100%';
+    }
 
-    var result = values.map(internalScope.offsetDistanceParse);
-    if (typeof result[0] === 'undefined' || typeof result[1] === 'undefined') {
+    if (second === 'top') {
+      second = '0%';
+    } else if (second === 'center') {
+      second = '50%';
+    } else if (second === 'bottom') {
+      second = '100%';
+    }
+
+    var leftValue = internalScope.offsetDistanceParse(first);
+    var topValue = internalScope.offsetDistanceParse(second);
+    if (typeof leftValue === 'undefined' || typeof topValue === 'undefined') {
       return undefined;
     }
+    return [leftValue, topValue];
+  }
 
-    return result;
+  function parseFourValuePosition (first, second, third, fourth) {
+    // Normalize so left|right appear first.
+    if (first === 'top' || first === 'bottom') {
+      if (!internalScope.isInArray(['left', 'right'], third)) {
+        return undefined;
+      }
+      // We received [ top | bottom ] <length-percentage> [ left | right ] <length-percentage>
+      return parseFourValuePosition(third, fourth, first, second);
+    }
+
+    // We have [ left | right ] <length-percentage> [ top | bottom ] <length-percentage>
+
+    var leftValue;
+    if (first === 'left') {
+      leftValue = internalScope.offsetDistanceParse(second);
+    } else if (first === 'right') {
+      leftValue = mirror(internalScope.offsetDistanceParse(second));
+    }
+
+    var topValue;
+    if (third === 'top') {
+      topValue = internalScope.offsetDistanceParse(fourth);
+    } else if (third === 'bottom') {
+      topValue = mirror(internalScope.offsetDistanceParse(fourth));
+    }
+
+    if (typeof leftValue === 'undefined' || typeof topValue === 'undefined') {
+      return undefined;
+    }
+    return [leftValue, topValue];
+  }
+
+  function mirror (distance) {
+    if (typeof distance === 'undefined') {
+      return undefined;
+    }
+    if (distance.unit !== '%') {
+      // We don't support calc expressions.
```

Comment:
"Implicit calc expression [right | bottom] <length>" not supported" would be a clearer comment here